pkg/plugin/GlobalPluginService.go

```diff
@@ -294,3 +301,1107 @@ func getVariableDto(pluginVariable *repository.PluginStepVariable) *PluginVariab
+			Index:                pluginStep.Index,
+			StepType:             pluginStep.StepType,
+			RefPluginId:          pluginStep.RefPluginId,
+			OutputDirectoryPath:  pluginStep.OutputDirectoryPath,
+			DependentOnStep:      pluginStep.DependentOnStep,
+			PluginPipelineScript: pluginScript,
+		}
+		pluginStepVariableResp := make([]*PluginVariableDto, 0)
+		for _, pluginStepVariable := range pluginStepVariables {
+			if pluginStepVariable.PluginStepId == pluginStep.Id {
+				pluginStepConditionDto := make([]*repository.PluginStepCondition, 0)
+				for _, pluginStepCondition := range pluginStepConditions {
+					if pluginStepCondition.ConditionVariableId == pluginStepVariable.Id {
+						pluginStepConditionDto = append(pluginStepConditionDto, &repository.PluginStepCondition{
+							Id:                  pluginStepCondition.Id,
+							PluginStepId:        pluginStepCondition.PluginStepId,
+							ConditionVariableId: pluginStepCondition.ConditionVariableId,
+							ConditionType:       pluginStepCondition.ConditionType,
+							ConditionalOperator: pluginStepCondition.ConditionalOperator,
+							ConditionalValue:    pluginStepCondition.ConditionalValue,
+							Deleted:             pluginStepCondition.Deleted,
+						})
+					}
+				}
+				pluginStepVariableResp = append(pluginStepVariableResp, &PluginVariableDto{
+					Id:                        pluginStepVariable.Id,
+					Name:                      pluginStepVariable.Name,
+					Format:                    pluginStepVariable.Format,
+					Description:               pluginStepVariable.Description,
+					IsExposed:                 pluginStepVariable.IsExposed,
+					AllowEmptyValue:           pluginStepVariable.AllowEmptyValue,
+					DefaultValue:              pluginStepVariable.DefaultValue,
+					Value:                     pluginStepVariable.Value,
+					VariableType:              string(pluginStepVariable.VariableType),
+					ValueType:                 pluginStepVariable.ValueType,
+					PreviousStepIndex:         pluginStepVariable.PreviousStepIndex,
+					VariableStepIndex:         pluginStepVariable.VariableStepIndex,
+					VariableStepIndexInPlugin: pluginStepVariable.VariableStepIndexInPlugin,
+					ReferenceVariableName:     pluginStepVariable.ReferenceVariableName,
+					PluginStepCondition:       pluginStepConditionDto,
+				})
+			}
+		}
+		pluginStepDto.PluginStepVariable = pluginStepVariableResp
+		pluginStepsResp = append(pluginStepsResp, pluginStepDto)
+	}
+	pluginMetadataResponse.PluginSteps = pluginStepsResp
+
+	return pluginMetadataResponse, nil
+}
+
+func (impl *GlobalPluginServiceImpl) deletePlugin(pluginDeleteReq *PluginMetadataDto, userId int32) (*PluginMetadataDto, error) {
+	dbConnection := impl.globalPluginRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return nil, err
+	}
+	// Rollback tx on error
+	defer tx.Rollback()
+	pluginMetaData, err := impl.globalPluginRepository.GetMetaDataByPluginId(pluginDeleteReq.Id)
```

Comment:
instead of directly starting deletion, let's check that if this plugin is referenced in some step or not. If it's being used we should throw error else ci/cd operations might break